python-avd/pyavd/_eos_designs/structured_config/underlay/dhcp_server.py

```diff
@@ -0,0 +1,114 @@
+# Copyright (c) 2023-2024 Arista Networks, Inc.
+# Use of this source code is governed by the Apache License 2.0
+# that can be found in the LICENSE file.
+from __future__ import annotations
+
+from functools import cached_property
+from ipaddress import AddressValueError, IPv4Address, ip_network
+
+from pyavd._utils import get
+
+from .utils import UtilsMixin
+
+
+class DhcpServerMixin(UtilsMixin):
+    """
+    Mixin Class used to generate structured config for one key.
+
+    Class should only be used as Mixin to a AvdStructuredConfig class.
+    """
+
+    _hostvars: dict
+
+    @cached_property
+    def _subnets(self) -> list:
+        """
+        Returns a list of dhcp subnets for downstream p2p interfaces.
+
+        Used for l3 inband ztp/ztr.
+        """
+        subnets = []
+        for peer in self._avd_peers:
+            peer_facts = self.shared_utils.get_peer_facts(peer, required=True)
+            for uplink in peer_facts["uplinks"]:
+                if (
+                    uplink["peer"] == self.shared_utils.hostname
+                    and uplink["type"] == "underlay_p2p"
+                    and uplink.get("ip_address")
+                    and "unnumbered" not in uplink["ip_address"]
+                    and get(peer_facts, "inband_ztp")
+                ):
+                    subnet = {
+                        "subnet": str(ip_network(f"{uplink['peer_ip_address']}/{uplink['prefix_length']}", strict=False)),
+                        "ranges": [{"start": str(uplink["ip_address"]), "end": str(uplink["ip_address"])}],
+                        "name": f"inband ztp for {peer}-{uplink['interface']}",
+                        "default_gateway": f"{uplink['peer_ip_address'].split('/')[0]}",
+                    }
+                    subnets.append(subnet)
+
+        return subnets
+
+    @cached_property
+    def _ipv4_ztp_boot_file(self) -> str | None:
+        """Returns the file name to allow for ZTP to CV."""
+        custom_bootfile = get(self._hostvars, "inband_ztp_bootstrap_file")
+        if custom_bootfile:
+            return custom_bootfile
+        cvp_instance_ips = get(self._hostvars, "cvp_instance_ips")
+        if not cvp_instance_ips:
+            return None
+        if "arista.io" in cvp_instance_ips[0]:
+            return "https://www.arista.io/ztp/bootstrap"
+
+        return f"https://{cvp_instance_ips[0]}/ztp/bootstrap"
```

Comment:
Redirection of the EoS running <4.30.0F to US-1A CVaaS (www.arista.io) would fail if switch is mapped to another region (only EOS >=4.30.0F has embedded support for redirector service). It might be better just to send it to whatever first CV FQDN/IP is configured under `cvp_instance_ips`?

```suggestion

        return f"https://{cvp_instance_ips[0]}/ztp/bootstrap"
```